torchgeo/datasets/loveda.py

```diff
@@ -0,0 +1,317 @@
+            length of dataset
+
+        """
+        return len(self.files)
+
+    def _load_files(self, scene_paths: List[str], split: str) -> List[Dict[str, str]]:
+        """Return the paths of the files in the dataset.
+
+        Args:
+            scene_paths: contains one or two paths, depending on whether user has
+                         specified only 'rural', 'only 'urban' or both
+            split: subset of dataset, one of [train, val, test]
+
+        """
+        images = []
+
+        for s in scene_paths:
+            images.extend(glob.glob(os.path.join(s, "images_png", "*.png")))
+
+        images = sorted(images)
+
+        if self.split != "test":
+            masks = [image.replace("images_png", "masks_png") for image in images]
+            files = [
+                dict(image=image, mask=mask) for image, mask, in zip(images, masks)
+            ]
+        else:
+            files = [dict(image=image) for image in images]
+
+        return files
+
+    def _load_image(self, path: str) -> Tensor:
+        """Load a single image.
+
+        Args:
+            path: path to the image
+
+        Returns:
+            tensor: the loaded image
+
+        """
+        filename = os.path.join(path)
+        with Image.open(filename) as img:
+            array = np.array(img.convert("RGB"))
+            tensor: Tensor = torch.from_numpy(array)  # type: ignore[attr-defined]
+            # Convert from HxWxC to CxHxW
+            tensor = tensor.permute((2, 0, 1))
+            return tensor
+
+    def _load_target(self, path: str) -> Tensor:
+        """Load a single mask corresponding to image.
+
+        Args:
+            path: path to the mask
+        Returns:
+            tensor: the mask of the image
+
+        """
+        filename = os.path.join(path)
+        with Image.open(filename) as img:
+            array = np.array(img.convert("L"))
+            tensor: Tensor = torch.from_numpy(
+                array
+            ).long()  # type: ignore[attr-defined]
+            return tensor
+
+    def _check_integrity(self) -> bool:
+        """Check the integrity of the dataset structure.
+
+        Returns:
+            True if the dataset directories and split files are found, else False
+
+        """
+        for s in self.scene_paths:
+            if not os.path.exists(s):
+                return False
+
+        return True
+
+    def _download(self) -> None:
+        """Download the dataset and extract it.
+
+        Raises:
+            AssertionError: if the checksum of split.py does not match
+
+        """
+        if self._check_integrity():
+            print("Files already downloaded and verified")
+            return
+
+        download_and_extract_archive(
+            self.url,
+            self.root,
+            filename=self.filename,
+            md5=self.md5 if self.checksum else None,
+        )
+
+    def plot(self, sample: Dict[str, Tensor], suptitle: Optional[str] = None) -> Figure:
```

Comment:
For consistency with other datasets, the `plot` method should only plot a single sample of image of image/mask, not a batch.